mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockResponse.java

```diff
@@ -78,7 +78,17 @@ public MockResponse setStatus(String status) {
 
   /** Returns the HTTP headers, such as "Content-Length: 0". */
   public List<String> getHeaders() {
-    return headers;
+    Headers headers = this.headers.build();
+    int size = headers.size();
+    List<String> headerList = new ArrayList<>(size);
+    for (int i = 0; i < size; i++) {
+      headerList.add(headers.name(i) + ": " + headers.value(i));
+    }
+    return headerList;
+  }
+
+  Headers getNewHeaders() {
```

Comment:
This will eventually replace `getHeaders()` but that requires a good bit of refactoring which will be in a follow-up.